modules/substrate/src/forks.rs

```diff
@@ -0,0 +1,508 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
```

Comment:
I guess we _could_, but does it provide any tangible benefit?

Reply:
Cleaner separation between production and test code imho. When I saw `forks.rs` I initially thought this module will be responsible for forks handling, and it turned out just to be a bunch of tests.

Reply:
That makes sense to me. I tried splitting it out and ran into a bunch of errors since I'm using a lot of private types in the tests. I'll look into it more tomorrow - but if I have to expose a bunch of private types to get the compiler happy I think I'll just leave it where it is (maybe with a rename to `fork_tests.rs`)

Reply:
I took another look, and it's a little gross to split into a proper integration test folder right now. I've opted to rename the module to `fork_tests.rs`. Pallets in Substrate typically have a `tests.rs` module within `src/` anyways, so I don't think it's that big a mental jump

Reply:
I was thinking about `src/tests` directory not top-level `tests` (meant for integraiton testing). `tests` module would have both `mock.rs` and `fork.rs` files. I'm okay with the prefix for now, but a bit afraid that this becomes a pattern and we will have a mix of test and src files directly in the `src` directory which would become messy.